### PR TITLE
Left-align stub arguments to dd (fixes #374)

### LINKED ISSUE #374: ape loader fails with busybox dd
in a dockerfile like so:
```Dockerfile
FROM alpine:3

RUN apk add git make bash zip
RUN git clone --depth=1 https://github.com/jart/cosmopolitan.git /build/cosmopolitan
RUN make -C /build/cosmopolitan -j -l4 -O O=o//tool/net/redbean.com
```
we see:
```
o//third_party/quickjs/qjsc.com.tmp.68677: exec: line 13: /build/cosmopolitan/o/tmp/ape-loader: not found

`make MODE= -j16 o//third_party/quickjs/qjscalc.c` exited with 127:
o//third_party/quickjs/qjsc.com -fbignum -o o//third_party/quickjs/qjscalc.c -c third_party/quickjs/qjscalc.js
consumed 634µs wall time
ballooned to 1,088kb in size
needed 595us cpu (0% kernel)
caused 133 page faults (100% memcpy)
7 context switch (100% consensual)
performed 160 read and 0 write i/o operations

make: *** [third_party/quickjs/quickjs.mk:134: o//third_party/quickjs/qjscalc.c] Error 127
make: Entering directory '/build/cosmopolitan'
make: Leaving directory '/build/cosmopolitan'
The command '/bin/sh -c make -C /build/cosmopolitan -j -l4 -O O=o//tool/net/redbean.com' returned a non-zero code: 2
```

poking around the build dir by hand we can uncover:
```
o//test/libc/release/smoke-nms.com.tmp.70227: exec: line 13: /build/cosmopolitan/o/tmp/ape-loader: not found

`make MODE= -j16 o//test/libc/release/smoke-nms.com.runs` exited with 127:
o//test/libc/release/smoke-nms.com
consumed 634µs wall time
ballooned to 1,100kb in size
needed 546us cpu (0% kernel)
caused 132 page faults (100% memcpy)
7 context switch (100% consensual)
performed 160 read and 0 write i/o operations

make: *** [build/rules.mk:76: o//test/libc/release/smoke-nms.com.runs] Error 127
make: Leaving directory '/build/cosmopolitan'
```

stracing the build with `strace -f -o /tmp/strace.txt make -C /build/cosmopolitan/ MODE= -j16 o//test/libc/release/smoke-nms.com.runs` shows:
```
/ # grep dd /tmp/strace.txt | tail -n 10
70257 set_tid_address(0x7ffb9a458f90)   = 70257
70257 stat("/usr/local/sbin/dd", 0x7ffc5f877c40) = -1 ENOENT (No such file or directory)
70257 stat("/usr/local/bin/dd", 0x7ffc5f877c40) = -1 ENOENT (No such file or directory)
70257 stat("/usr/sbin/dd", 0x7ffc5f877c40) = -1 ENOENT (No such file or directory)
70257 stat("/usr/bin/dd", 0x7ffc5f877c40) = -1 ENOENT (No such file or directory)
70257 stat("/sbin/dd", 0x7ffc5f877c40)  = -1 ENOENT (No such file or directory)
70257 stat("/bin/dd", {st_mode=S_IFREG|0755, st_size=824984, ...}) = 0
70259 execve("/bin/dd", ["dd", "if=o//test/libc/release/smoke-nm"..., "of=/build/cosmopolitan/o/tmp/ape"..., "skip=    3200", "count=      28", "bs=64"], 0x7ffb9a3be2f8 /* 12 vars */) = 0
70259 set_tid_address(0x7f58fed0bf90)   = 70259
70259 write(2, "dd: invalid number '    3200'\n", 30) = 30
```

if i `apk add coreutils`, it succeeds.

i looked at the code in ape/ape.S, but i do not understand where `.shstub` assembly directive comes from, but it seems that busybox dd doesnt like spaces in number arguments, like
```
/ # dd 'skip=       1'
dd: invalid number '       1'
```

can we fix the script in such a way that it is compatible with busybox dd?

### PR DESCRIPTION
**Pull request by notwa**
This satisfies busybox's non-standard integer argument parsing, and even saves a few bytes.

since this technically changes APE itself, I'd prefer to have tested this on every targeted platform, but I don't have the necessary VMs configured. the GitHub workflow checks should pass, but that doesn't really mean anything. the bootstrap files should be updated eventually.

### CODE REVIEW

**Comment by notwa** [comment]
oh shoot, I *just* noticed a bug. don't merge this quite yet.

**Comment by notwa** [comment]
a little evil, but it's fixed.

tests: https://eaguru.guru/t/bcd-test.c

File: ape/macros.internal.h [hunk 1]
```diff
@@ -221,44 +221,41 @@
...
+  HIDDEN(SYM##_bcs1 = BCD_RIGHT(X))
+#define BCD_SMEAR(X) ((X) + (X) * 10000)
+#define BCD_LEFT(X)                                      \
+  (((X)) < 10000     ? BCD_RIGHT(BCD_SMEAR(X)) | 0x10    \
+   : (X) < 100000    ? BCD_RIGHT(BCD_SMEAR((X) / 10))    \
+   : (X) < 1000000   ? BCD_RIGHT(BCD_SMEAR((X) / 100))   \
+   : (X) < 10000000  ? BCD_RIGHT(BCD_SMEAR((X) / 1000))  \
+   : (X) < 100000000 ? BCD_RIGHT(BCD_SMEAR((X) / 10000)) \
+                     : 0xffffffffffffffff)
+#define BCD_RIGHT(X) \
```
Comment (jart) [inline ape/macros.internal.h:242]: I love the style here.

**Review by jart — APPROVED** [review]
Looks great. It's really exciting that we're going to be able to support BusyBox. Thank you!

**Pull request merged**